### PR TITLE
G signal

### PR DESCRIPTION
**Pull request by gharazka**
Redirect input
Redirect output
Here_doc
Append output
Quotes handled

Bugs:
unclosed quotes sometimes go into infinite loop
redirecting output and appending to multiple files doesn't work
redirecting output without a command at the start doesn't work

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -55,17 +62,19 @@ int	main(int _argc, char *_argv[], char *envp[])
 		if (ft_strlen(line) > 0)
 		{
 			add_history(line);
-			id = fork();
-			split_line = ft_split(line, '|');
 			t_list *tokens = tokenize_input(line);
+			id = fork();
 			if (id == 0)
 			{
+				for (t_list *temp = tokens; temp != NULL; temp = temp->next)
```
Comment (nuketayev) [inline src/main.c:69]: What is the purpose of temp?

Comment (gharazka) [reply src/main.c:69]: If we use the original linked list then we don't have a pointer to the start of it, therefore we need a temp so we can print all the values (for testing) and then still use the original linked list to execute the commands

Comment (nuketayev) [reply src/main.c:69]: oh I see it now

File: src/parse_input.c [hunk 2]
```diff
@@ -9,16 +9,16 @@ static int	count_slash_semicolon(char *line)
 		return (0);
 	i = 0;
 	count = 0;
-	while (line[i])
+	while (line[i] && line[i] != until)
```
Comment (nuketayev) [inline src/parse_input.c:12]: #note for myself. parentaces for better reading

File: src/parse_input.c [hunk 3]
```diff
@@ -27,45 +27,28 @@ static char	*remove_slash_semicolon(char *line)
...
-	f_line = (char *)malloc(sizeof(char) * (ft_strlen(line) - count + 1));
+	count = 2;
+	f_line = (char *)malloc(sizeof(char) * (ft_strlen(line) - 1));
 	i = 0;
 	j = 0;
-	while (line[i])
+	while (line[i] && count > 0)
 	{
-		if (line[i] != '\\' && line[i] != ';')
+		if (line[i] != c)
```
Comment (nuketayev) [inline src/parse_input.c:36]: !<= ?

Comment (gharazka) [reply src/parse_input.c:36]: ?

File: src/parse_input.c [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #include "../inc/minishell.h"
 
-static int	count_slash_semicolon(char *line)
+static int	count_char_until(char *line, char c, char until)
```
Comment (nuketayev) [inline src/parse_input.c:3]: #note for me. count_the_char

Comment (gharazka) [reply src/parse_input.c:3]: This can be deleted, I decided to not use it afterwards

Comment (nuketayev) [reply src/parse_input.c:3]: I noticed it

**Review by nuketayev — APPROVED** [review]

**Pull request merged**